lib/ansible/plugins/connection/vmware_tools.py

```diff
@@ -0,0 +1,452 @@
+    options:
+      vmware_host:
+        description:
+          - Address for the connection (vCenter or ESXi Host).
+        env:
+          - name: VI_SERVER
+        vars:
+          - name: ansible_host
+          - name: ansible_vmware_host
+        required: True
+      vmware_user:
+        description:
+          - Username for the connection.
+          - "Requires the following permissions on the VM:
+               - VirtualMachine.GuestOperations.Execute
+               - VirtualMachine.GuestOperations.Modify
+               - VirtualMachine.GuestOperations.Query"
+        env:
+          - name: VI_USERNAME
+        vars:
+          - name: ansible_user
+          - name: ansible_vmware_user
+        required: True
+      vmware_password:
+        description:
+          - Password for the connection.
+        env:
+          - name: VI_PASSWORD
+        vars:
+          - name: ansible_password
+          - name: ansible_vmware_password
+        required: True
+      validate_certs:
+        description:
+          - Verify SSL for the connection.
+          - "Note: This will validate certs for both C(vmware_host) and the ESXi host running the VM."
+        vars:
+          - name: ansible_vmware_validate_certs
+        default: True
+        type: bool
+      silence_tls_warnings:
+        description:
+          - Don't output warnings about insecure connections.
+        vars:
+          - name: ansible_vmware_silence_tls_warnings
+        default: False
+        type: bool
+      vm_path:
+        description:
+          - VM path relative to the connection.
+          - "vCenter Example: C(Datacenter/vm/Discovered virtual machine/testVM) (Needs to include C(vm) between the Datacenter and the rest of the VM path.)"
+          - "ESXi Host Example: C(ha-datacenter/vm/testVM) (Needs to include C(vm) between the Datacenter (C(ha-datacenter)) and the rest of the VM path.)"
+          - Must include VM name, appended to 'folder' as would be passed to vmware_guest.
+        vars:
+          - name: ansible_vmware_guest_path
+        required: True
+      vm_user:
+        description:
+          - VM username.
+        vars:
+          - name: ansible_user
+          - name: ansible_vmware_tools_user
+        required: True
+      vm_password:
+        description:
+          - VM password.
+        vars:
+          - name: ansible_password
+          - name: ansible_vmware_tools_password
+        required: True
+      exec_command_sleep_interval:
+        description:
+          - exec command sleep interval in seconds.
+        vars:
+          - name: ansible_vmware_tools_exec_command_sleep_interval
+        default: 0.5
+        type: float
+      file_chunk_size:
+        description:
+          - File chunk size.
+          - "(Applicable when writing a file to disk, example: using the C(fetch) module.)"
+        vars:
+          - name: ansible_vmware_tools_file_chunk_size
+        default: 128
+        type: integer
```

Comment:
Also add parameter `VMWARE_PORT`. Sometime vCenter is located on custom port rather than `443`. 

https://github.com/ansible/ansible/blob/a2eb227970b5f53a648af3daf58d745d2f33067e/lib/ansible/plugins/doc_fragments/vmware.py#L85

Reply:
Yeah, would be good to add... do you immediately know how it would be passed?

Reply:
It is available in PR.

Reply:
nevermind, I see that you added it